### PR TITLE
Add follow-on fields to mobile work order close form

### PR DESCRIPTION
**Pull request by LBHCallumM**
## Summary of Changes

Add Follow-on screen to mobile CloseWorkOrder screen. 

- Update 'Completed' with 'Visit completed'
- Conditionally send `followOnRequest` data in request
- Update success modal message from 'completed' to 'closed'
- Re-enabled old tests which had been disabled

### Close button (unchanged)
image

### New fields
image

### Button text updates
image

### Follow-on details page
image

### Additional trades
image

### Conditionally required materials description
image

### CODE REVIEW

File: src/components/WorkOrder/MobileWorkingWorkOrderView.js [hunk 1]
```diff
@@ -118,29 +122,49 @@ const MobileWorkingWorkOrderView = ({ workOrderReference }) => {
   const onWorkOrderCompleteSubmit = async (data) => {
     setLoading(true)
 
+    let followOnRequest = null
+
+    if (data['followOnStatus'] === 'furtherWorkRequired') {
```
Comment (adamtry) [inline src/components/WorkOrder/MobileWorkingWorkOrderView.js:127]: Why not `data.followOnStatus` with dot notation etc, and is there any risk of null/undefined here that should be handled?

Comment (LBHCallumM) [reply src/components/WorkOrder/MobileWorkingWorkOrderView.js:127]: The backend request object doesnt like null. The object must either be populated, or not present in the request

Comment (LBHCallumM) [reply src/components/WorkOrder/MobileWorkingWorkOrderView.js:127]: Ill change to dot notation if ther are other changes. Im not worth waiting another 20 mins for the tests to run

File: src/utils/hact/workOrderComplete/closeWorkOrder.js [hunk 2]
```diff
@@ -26,6 +27,36 @@ export const buildCloseWorkOrderData = (
...
   }
+
+  if (followOnRequest !== null) {
+    dataObject['followOnRequest'] = followOnRequest
+  }
+
+  return dataObject
+}
+
+export const buildFollowOnRequestData = (
```
Comment (adamtry) [inline src/utils/hact/workOrderComplete/closeWorkOrder.js:38]: Is this just to help with enforcing the schema?

Comment (LBHCallumM) [reply src/utils/hact/workOrderComplete/closeWorkOrder.js:38]: Yeah, makes it more explicit given there is no typescript

File: cypress/integration/work-order/attend/close.spec.js [hunk 3]
```diff
@@ -77,11 +77,235 @@ describe('Closing my own work order', () => {
     cy.loginWithOperativeRole()
   })
 
-  context.only('during normal working hours', () => {
+  context('during normal working hours', () => {
     beforeEach(() => {
       cy.clock(new Date(now).setHours(12, 0, 0))
     })
 
+    it('shows a validation error when no reason is selected', () => {
```
Comment (adamtry) [inline cypress/integration/work-order/attend/close.spec.js:85]: For E2E tests, I'd suggest combining the validation error test cases into a single case for showing appropriate validation errors. These seem too focused for e2e tests

**Review by adamtry — APPROVED** [review]

**Pull request merged**